### DISCUSSION TITLE
Notifier and AsyncNofitier does not have keep alive property

### DISCUSSION THREAD

**Discussion by Velkamhell97** [rrousselGit/riverpod#2586] [Q&A] [answered]
The `Notifier` and `AsyncNotifier` providers do not have the `keep alive` property to maintain the state, are these providers `autodispose` by default ?

**Comment by AhmedLSayed9** [comment, marked as answer]
Notifier and AsyncNotifier both can be annotated as a `keepAlive` and the manual keep alive on condition `ref.keepAlive` can also be used inside their build.

**Reply by Velkamhell97** [reply]
Can you show me an example of how to set this property manually ? the editor dont show me this property for the `ref object`

**Reply by AhmedLSayed9** [reply]
```dart
@riverpod
class SomeNotifier extends _$SomeNotifier {
  @override
  Something build(){
    ref.keepAlive();
    return Something;
  }
}
```

**Reply by AhmedLSayed9** [reply]
Note: You must have the generated code first before you can use ref.something():

**Reply by Velkamhell97** [reply]
Oh I was referring to the way where no code generation is used, but I know what the problem was, I had not read the documentation, there it is indicated that the keep alive is a property only available for autodispose providers, I had to change my `AsyncNotifier` for an `AutoDisposeAsyncNotifier`. But anyway thanks for your reply 👏

**Reply by AhmedLSayed9** [reply]
Yeah that's true. There's no point in using keepAlive in an AlwaysAlive provider :)
